components/kubernetes-api/src/main/java/io/fabric8/kubernetes/api/KubernetesHelper.java

```diff
@@ -1284,6 +1296,45 @@ public static String getServiceURL(KubernetesClient client, String serviceName,
         String portalIP = srv.getSpec().getPortalIP();
         Integer portNumber = port.getPort();
         if (Strings.isNullOrBlank(portalIP)) {
+            IngressList ingresses = client.extensions().ingresses().inNamespace(serviceNamespace).list();
+            if (ingresses != null) {
+                List<Ingress> items = ingresses.getItems();
+                if (items != null) {
+                    for (Ingress item : items) {
+                        IngressSpec spec = item.getSpec();
+                        if (spec != null) {
+                            List<IngressRule> rules = spec.getRules();
+                            if (rules != null) {
+                                for (IngressRule rule : rules) {
+                                    HTTPIngressRuleValue http = rule.getHttp();
+                                    if (http != null) {
+                                        List<HTTPIngressPath> paths = http.getPaths();
+                                        if (paths != null) {
+                                            for (HTTPIngressPath path : paths) {
+                                                IngressBackend backend = path.getBackend();
+                                                if (backend != null) {
+                                                    String backendServiceName = backend.getServiceName();
+                                                    if (serviceName.equals(backendServiceName) && portsMatch(port, backend.getServicePort())) {
+                                                        String answer = rule.getHost();
+                                                        if (Strings.isNotBlank(answer)) {
+                                                            // TODO add a path too?
+                                                            String pathPostfix = path.getPath();
+                                                            if (Strings.isNullOrBlank(pathPostfix)) {
+                                                                pathPostfix = "/";
+                                                            }
+                                                            return "http://" + URLUtils.pathJoin(answer, pathPostfix);
```

Comment:
ah - spotted the top level 'tls' property!
http://kubernetes.io/docs/user-guide/ingress/